From c26f1f39b4371605ebf0ae5c5dacbdb2ad142ccd Mon Sep 17 00:00:00 2001
From: Liu Hua <weldonliu@tencent.com>
Date: Mon, 1 Nov 2021 20:50:58 +0800
Subject: [PATCH 039/245] cr-dump: fail dumping when zombie process with sid 0

A zombie process with 0 sid has a session leader in
outer pidns and has ignored SIGHUP. Criu has no idea
to restore this type of process, so fail the dumpping.

Signed-off-by: Liu Hua <weldonliu@tencent.com>
---
 criu/cr-dump.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/criu/cr-dump.c b/criu/cr-dump.c
index ecc99f116..c972e343a 100644
--- a/criu/cr-dump.c
+++ b/criu/cr-dump.c
@@ -1129,6 +1129,13 @@ static int dump_zombies(void)
 		item->pgid = pps_buf.pgid;
 
 		BUG_ON(!list_empty(&item->children));
+
+		if (!item->sid) {
+			pr_err("A session leader of zombie process %d(%d) is outside of its pid namespace\n",
+			       item->pid->real, vpid(item));
+			goto err;
+		}
+
 		if (dump_one_zombie(item, &pps_buf) < 0)
 			goto err;
 	}
-- 
2.35.1

